sentry_sdk/scope.py

```diff
@@ -1250,3 +1421,55 @@ def __repr__(self):
             hex(id(self)),
             self._name,
         )
+
+
+@contextmanager
+def new_scope():
+    # type: () -> Generator[Scope, None, None]
+    """
+    Context manager that forks the current scope and runs the wrapped code in it.
+
+    .. versionadded:: X.X.X
+    """
+    current_scope = Scope.get_current_scope()
+    forked_scope = current_scope.fork()
+    token = _current_scope.set(forked_scope)
```

Comment:
`m` instead of holding on to the token, could we simply hold a reference to the scope we want to restore later on? Same question applies to `isolated_scope()` below.

Reply:
Yes. The token business is just how context vars in Python work. (Internally Python will hold a reference to the scope I guess) So whatever works in Java is OK, token is not a hard requirement.